cli/src/main.rs

```diff
@@ -52,6 +52,225 @@ pub fn main() {
         }
     };
 
+    let regtest = matches.is_present("regtest");
+    if regtest {
+        use std::ffi::OsString;
+        use std::fs::File;
+        use std::io;
+        use std::path::PathBuf;
+        use std::process::{Command, Stdio};
+        use std::{thread, time};
+
+        // confirm we are in running from zingolib as an anchor for our directory context
+        let revparse = Command::new("git")
+            .args(["rev-parse", "--show-toplevel"])
+            .output()
+            .expect("no git!? time to quit.");
```

Comment:
No more rhymes now!  I mean it!

Reply:
Nice defensive check.

Reply:
As I read further I am realizing that `revparse` is the `ident` for a variable that specifies the top-level directory of the repository.   Is that correct?  In that case let's name it something like:

`toplevel_dir_nym` <-- emphasizes its relationship to the source code
OR
`workspace_dir_nym` <-- emphasizes its relationship to a Rust project layout

either of those `ident`s provide the reader with important semantic information, whereas `revparse` reports on an implementation detail of _how_ that information was acquired.

Reply:
> No more rhymes now! I mean it!

+1 on rhymes and fun ;9